pkg/util/wg.go

```diff
@@ -0,0 +1,22 @@
+package util
+
+import (
+	"context"
+	"sync"
+)
+
+func StartInWaitGroup(f func(), group *sync.WaitGroup) {
+	group.Add(1)
+	go func() {
+		defer group.Done()
+		f()
+	}()
+}
+
+func StartInWaitGroupWithContext(ctx context.Context, f func(ctx context.Context), group *sync.WaitGroup) {
+	group.Add(1)
```

Comment:
Can be shortened to `StartInWaitGroup(func() { f(ctx) }, group)`

Reply:
In general we need to pass the context so the function f can later be invoked with it or do we expect to be able to use the context from closure ?